rootfs/etc/nginx/lua/defer_to_timer.lua

```diff
@@ -0,0 +1,39 @@
+local queue = {}
+local MAX_QUEUE_SIZE = 10000
+local FLUSH_INTERVAL = 1
+
+local _M = {}
+
+local function flush_queue()
+  local current_queue = queue
+  queue = {}
+
+  for _,v in ipairs(current_queue) do
+    v.func(unpack(v.args))
+  end
+end
+
+function _M.init_worker()
+  local _, err = ngx.timer.every(FLUSH_INTERVAL, flush_queue)
+  if err then
+    ngx.log(ngx.ERR, string.format("error when setting up timer.every for flush_queue: %s", tostring(err)))
+  end
+end
+
+function _M.enqueue(func, ...)
+  if #queue >= MAX_QUEUE_SIZE then
+    return "deferred timer queue full"
+  end
+
+  table.insert(queue, { func = func, args = {...} })
+
+  return nil
```

Comment:
Nitpick: you don't need the `return nil` here, though I'm assuming you just did it to be explicit.

Reply:
looks like resty-core-* modules stick to being implicit in these cases - I'll stick to that.